### ISSUE TITLE
Delete the __main__ in twisted/protocols/shoutcast.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3714]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3714|
|Type|enhancement|
|Created|2009-03-30 19:01:41Z|
|Branch|https://github.com/twisted/twisted/tree/shoutcast-main-3714|

This shouldn't be a script.  We can add a separate client that does shoutcast stuff later if we want.

Attachments:

* [shoutcast.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/1fc/1fc43d0b8d219ec2cd84d919f7b75c7993148026/e40f565b8d1e235f23a956eec641ad8526b982a6.patch) (658 bytes) - added by cary on 2009-03-30 21:50:54Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3714 3714
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_shoutcast_main_3714 branches/shoutcast-main-3714
branch_author__cary__thijs cary, thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1238439701000000 1238439701000000
changetime__1238518241000000 1238518241000000
version__None None
owner__ 
cc__thijs cc__radix
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="cary's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|cary removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

Thanks for the patch. I think it could be used as an example if there's not one for shoutcast already.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [26549]) Branching to 'shoutcast-main-3714'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [26550]) Apply shoutcast.patch refs [#3714](https://github.com/twisted/twisted/issues/3714)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

(In [26551]) Add shoutcast client example refs [#3714](https://github.com/twisted/twisted/issues/3714)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @exarkun|
|-|-|

Looks good to me

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

Oops.  Actually, shoutcast depends on twisted.web!  I guess I'll merge this anyway, since it's still an improvement, but we're going to need to move shoutcast into the Twisted Web package (module moved for split ha ha).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [26572]) Merge shoutcast-main-3714 

Author: cary, thijs
Reviewer: exarkun
Fixes: [#3714](https://github.com/twisted/twisted/issues/3714)

Move the `__main__` out of `twisted/protocols/shoutcast.py` and into an example script.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
